src/core/geometry/qgscircularstring.cpp

```diff
@@ -317,7 +317,14 @@ bool QgsCircularString::fromWkt( const QString &wkt )
   if ( parts.second.compare( QLatin1String( "EMPTY" ), Qt::CaseInsensitive ) == 0 )
     return true;
 
-  setPoints( QgsGeometryUtils::pointsFromWKT( parts.second, is3D(), isMeasure() ) );
+  if ( parts.second.compare( QLatin1String( "NULL" ), Qt::CaseInsensitive ) == 0 )
+    return false;
+
+  QgsPointSequence points = QgsGeometryUtils::pointsFromWKT( parts.second, is3D(), isMeasure() );
+  if ( points.isEmpty() )
```

Comment:
Unless I'm reading this wrong, this would prevent creation of geometries from `CircularString ()`. We should be forgiving here and treat things like `CircularString ()` as `CircularString( EMPTY )` (same with NULL). 

The QGIS WKT parser was designed to be as forgiving as possible, rather then strictly follow the standards. It should consume any standard WKT string, but if we can intrepret a malformed string then we need to allow this.

Reply:
yes with my PR `CircularString ()` is unallowed you have to write `circularstring EMPTY` (as on postgis).

The only "non standard" is `LINESTRING Z ...` instead of `LINESTRINGZ` (as on postgis too).

Reply:
Can you revert that change? It was an intentional choice to be forgiving...

Reply:
For me, a malformed string sould not be allowed. But `CircularString ( )` could be an exception, and can return an empty geometry.

Reply:
Why would we want to be restrictive though? If you get a dataset with malformed wkt, would you prefer to be able to import directly in qgis or spend ages mucking around and manually fixing it? We should definitely be as permissive as possible on import, and as strict as possible on output...

Reply:
Personally, I prefer to have a strict behavior indicating an error in the formatting of my string than having a tool assuming a result for me (especially when that result is wrong, like today.)

But I agree with you that it's not pleasant to have to manually fix strings that have errors.

So, what can we do to fix this parser?

We've already agreed on those (even if I haven't pushed my commit yet):
LineString () => LineString EMPTY
LineString ( EMPTY ) = > LineString EMPTY

But what about this WKT
Point (( 9, 9)) => POINT (9 9) or NULL ?
LINESTRING (( 1 1, 22, 3 3) => ?
LINESTRING (( 1 1, 2 2, 3 3)) => ?

Thanks.

Reply:
> Personally, I prefer to have a strict behavior indicating an error in the formatting of my string than having a tool assuming a result for me (especially when that result is wrong, like today.)

Yeah, but you're coming from a developer's point of view :laughing:  End users don't care about this -- they just want the data imported with as little messing around as possible!

> Point (( 9, 9)) => POINT (9 9) or NULL

I'd say POINT (9 9) -- the intention is clear here

> LINESTRING (( 1 1, 22, 3 3)

This one should error -- it's completely unclear what the intention was

> LINESTRING (( 1 1, 2 2, 3 3)) 

We should just ignore the extraneous brackets, and treat it as 
LINESTRING ( 1 1, 2 2, 3 3)

Reply:
> Yeah, but you're coming from a developer's point of view laughing End users don't care about this -- they just want the data imported with as little messing around as possible!

I was (and I'm still) an user before being a developper :wink: but I've always liked when software returns results as strict as possible -- maybe I'm crazy :laughing:  

Whatever I've no objection to do these enhancements. I'm just confused since I'll have not the time to do this before the release. 

As you want, you merge it now [1] and I add modifications next days or we postpone this bugfix and it will backported.

[1] ATM, except "geometry()" which returns an empty geometry and which I constrained, others return false geometry "Point( (1 1)" returns "Point( 0 1)

Reply:
We definitely should defer -- if we merge in the current strict approach then it's quite likely to break some existing workflows for users